spec/helpers/patient_notifications_helper_spec.rb

```diff
@@ -0,0 +1,173 @@
+      let(:invalidated_at) { 2.days.ago }
+
+      before { patient.update!(invalidated_at:) }
+
+      it "returns invalidated notification" do
+        expect(notifications.count).to eq(1)
+        expect(notifications.first).to include(
+          date_time: invalidated_at,
+          message: "Record flagged as invalid"
+        )
+      end
+    end
+
+    context "when patient is restricted" do
+      let(:restricted_at) { 3.days.ago }
+
+      before { patient.update!(restricted_at:) }
+
+      it "returns restricted notification" do
+        expect(notifications.count).to eq(1)
+        expect(notifications.first).to include(
+          date_time: restricted_at,
+          message: "Record flagged as sensitive"
+        )
+      end
+    end
+
+    context "when patient has gillick no notify vaccination records" do
+      let(:created_at) { 4.days.ago }
+
+      let(:vaccination_record) do
+        create(
+          :vaccination_record,
+          patient: patient,
+          programme: programme,
+          notify_parents: false,
+          created_at:
+        )
+      end
+
+      before { vaccination_record }
+
+      it "returns gillick no notify notification" do
+        expect(notifications.count).to eq(1)
+        notification = notifications.first
+        expect(notification[:date_time]).to eq(created_at)
+        expect(notification[:message]).to include(
+          "Child gave consent for HPV vaccination under Gillick competence"
+        )
+        expect(notification[:message]).to include(
+          "does not want their parents to be notified"
+        )
+      end
+    end
+
+    context "when patient has multiple vaccination records with the same notify_parents values" do
+      let(:other_programme) { create(:programme, :flu) }
+
+      let(:notify_record) do
+        create(
+          :vaccination_record,
+          patient:,
+          programme: other_programme,
+          notify_parents: false
+        )
+      end
+      let(:no_notify_record) do
+        create(:vaccination_record, patient:, programme:, notify_parents: false)
+      end
+
+      before do
+        notify_record
+        no_notify_record
+      end
+
+      it "only includes records with notify_parents false in the message" do
+        expect(notifications.count).to eq(1)
+        expect(notifications.first[:message]).to include(
+          "Flu and HPV vaccinations"
+        )
+      end
+    end
+
+    context "when patient has multiple vaccination records with different notify_parents values" do
+      let(:other_programme) { create(:programme, :flu) }
+
+      let(:notify_record) do
+        create(
+          :vaccination_record,
+          patient:,
+          programme: other_programme,
+          notify_parents: true
+        )
+      end
+      let(:no_notify_record) do
+        create(:vaccination_record, patient:, programme:, notify_parents: false)
+      end
+
+      before do
+        notify_record
```

Comment:
You can use `let!` to avoid the need to do this, e.g. `let!(:notify_record)`. However, if the values aren't being used in the test it's okay to put them in a `before` block without a name.